src/lib.rs

```diff
@@ -100,8 +102,29 @@ pub trait Core: Send + Sync {
         completion: ActivityTaskCompletion,
     ) -> Result<(), CompleteActivityError>;
 
-    /// Indicate that a long running activity is still making progress
-    async fn send_activity_heartbeat(&self, task_token: ActivityHeartbeat) -> Result<(), ()>;
+    /// Notify workflow that activity is still alive. Long running activities that take longer than
+    /// `activity_heartbeat_timeout` to finish must call this function in order to report progress,
+    /// otherwise activity will timeout and new attempt will be scheduled.
+    /// `result` contains latest known activity cancelation status.
+    /// Note that heartbeat requests are getting batched and are sent to the server periodically,
+    /// this function is going to return immediately and cancelation status will be taken from the server response
+    /// to the most recently sent batch.
```

Comment:
Should we even bother with this? Do the existing SDKs batch? In thinking about it, it seems totally reasonable to expect the activity code to block briefly while it does this. It also gives you a reason *not* to do it in a tight loop, etc, which is probably a good thing. We get to avoid the complexity of this mechanism too.

Reply:
Discussed offline, summary: we are going to decouple heartbeats and cancellation in the long run. Heartbeat reporting should be fast and can be batched by the core.